### ISSUE TITLE
TGMapViewController not deallocated on iOS

### ISSUE THREAD

**Issue by pierre-racine** [tangrams/tangram-es#1690]
Hello,

I have a memory problem with TGMapViewController.
If I put a TGMapViewController in the stack of navigation controller and after I pop it, the TGMapViewController is not deallocated.

TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:
I took the example project and I added a navigation controller. When i pop the dealloc method is not called and the memory remains high.

Does anyone have a solution for this?

Thank you !

**Comment by karimnaaji** [comment]
Are you building using ARC? Can you provide what container is being used for the stack (or code sample for that)? It is possible that the data structure you are using holds the memory for the map view.

**Comment by pierre-racine** [comment]
Yes, I use ARC. It's a basic project : 
-UINavigationController
-UIViewController (RootViewController)
-> show (with UIButton on storyboard)
-TGMapViewController (GLKViewController) with this code : 
 
```
-(void)viewDidLoad
{
   [super viewDidLoad];
}

-(void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    NSURL * sceneURL = [NSURL URLWithString:@"https://tangrams.github.io/walkabout-style/walkabout-style.yaml"];
    [self loadSceneAsyncFromURL:sceneURL withUpdates:@[[[TGSceneUpdate alloc]initWithPath:@"global.sdk_mapzen_api_key" value:@"MY_API_KEY"]]];
}

- (IBAction)back:(id)sender {
    [self.navigationController popViewControllerAnimated:YES];
}
```

If i pop TGMapViewController the memory remains high.

It's Tangram-es 0.8.1 install with CocoaPods.

I put the test project as an attachment : 
[tangram_test.zip](https://github.com/tangrams/tangram-es/files/1388430/tangram_test.zip)

Thank you

**Comment by karimnaaji** [comment]
Thanks, having a code sample is really helpful, I have been able to reproduce and fix the issue. 

We introduced this bug in some some early updates when modernizing the objective-c layer for this platform, it happens that we are not tearing down the native resources of the map. Part of this change will be reverted and we will make sure that this issue is fixed in the next release or patch release.
